File: android/app/src/main/java/drop/mesh/data/PendingPairing.kt
package drop.mesh.data

import java.util.UUID

/** 等待用户决定的入站连接。 */
data class PendingPairing(
    val id: UUID = UUID.randomUUID(),
    val peer: Device,
    val receivedAt: Long = System.currentTimeMillis(),
)

enum class PairingDecision { REJECT, ALLOW_ONCE, TRUST }

File: windows/ShareWindows/Models/PendingPairing.cs
using System;

namespace MeshDrop.Models;

public sealed record PendingPairing(Guid Id, Device Peer, DateTime ReceivedAt)
{
    public static PendingPairing Create(Device peer) => new(Guid.NewGuid(), peer, DateTime.Now);
}

public enum PairingDecision { Reject, AllowOnce, Trust }
